### PR TITLE
check-style: a reincarnation of check-style script

### PR DESCRIPTION
**Pull request by dorileo**
v2:
   + added colorized output;
   + removed a left over import statement;

New scripts "looks" more robust and extensible, also adds a new
argument -r <refspec> where the user may provide a refspec and not
simply a base commit. Everything else remain the same - besides the
scripts file extension, since it's now written in python.

Signed-off-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,4 @@
 /sol_flow_modules.mk
 /sol_linux_micro_modules.mk
 /sol_pin_mux_modules.mk
+*.uncrustify
```
Comment (barbieri) [inline .gitignore]: Missing newline?

File: tools/check-style.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return output
+    except subprocess.CalledProcessError as e:
+        try:
+            print("%s%s" % (msg_prefix, e.output))
+        except BrokenPipeError as e:
+            return None
+        return None
+
+def check_diff():
+    diff = which("diff")
```
Comment (barbieri) [inline tools/check-style.py]: Hum, why not use difflib directly?
https://docs.python.org/3.4/library/difflib.html

File: .gitignore (hunk 1 above)
Comment (dorileo) [inline .gitignore]: ok

File: tools/check-style.py (hunk 2 above)
Comment (dorileo) [inline tools/check-style.py]: seems a good thing...

**Comment by dorileo** [comment]
new version coming....

**Pull request closed without merging**